### PR TITLE
Document for semidefinite programming

### PR DESCRIPTION
**Pull request by pizhn**
Document for semidefinite programming & make some interface names consistent

### CODE REVIEW

**Comment by mohdibntarek** [comment]
I changed the structure of the Nonconvex package removing all the code from it and turning it into a shell package. The documentation will live here but the algorithms and wrappers live in their individual packages. Your code is in NonconvexSearch and NonconvexSemidefinite. Also NonconvexMultistart wraps Hyperopt. Please refactor this PR to match the pattern in the current master branch docs. Thanks.

File: docs/src/algorithms/sdp.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Semidifinite programming
+
+## Description
+
+If you have matrix in your optimization objective and you need to keep that positive semidefinite, Nonconvex supports a series of interfaces for [Barrier method for semidefinite programming](http://eaton.math.rpi.edu/faculty/Mitchell/courses/matp6640/notes/24A_SDPbarrierbeamer.pdf), which is implemented as a meta-algorithm transferring your optimization target to a general linear programming problem, then solving by pre-specified `sub_options`.
```
Comment (matbesancon) [inline docs/src/algorithms/sdp.md]:
```suggestion
If you have a matrix in your optimization problem that needs to remain positive semidefinite, Nonconvex supports a series of interfaces for the [Barrier method for semidefinite programming](http://eaton.math.rpi.edu/faculty/Mitchell/courses/matp6640/notes/24A_SDPbarrierbeamer.pdf), which is implemented as a meta-algorithm transferring your optimization target to a general linear programming problem, then solving by pre-specified `sub_options`.
```

File: docs/src/algorithms/sdp.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+## Quick start
+
+Optimizating over a multivariate gaussian distribution with artificial samples using `Ipopt`:
```
Comment (matbesancon) [inline docs/src/algorithms/sdp.md]:
```suggestion
Optimizing over a multivariate Gaussian distribution with artificial samples using `Ipopt`:
```

File: docs/src/algorithms/sdp.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
```
Comment (matbesancon) [inline docs/src/algorithms/sdp.md:10]: You don't need to indent inside a code block

**Pull request merged**